src/pages/steps/EventTypeAndThemeStep.tsx

```diff
@@ -144,11 +212,34 @@ const EventTypeAndThemeStep = ({
     scope,
   });
 
-  const types = getTypesByScopeQuery.data ?? [];
+  const types = useMemo(
+    () => getTypesByScopeQuery.data ?? [],
+    [getTypesByScopeQuery.data],
+  );
 
-  const themes =
-    types?.find((type) => type.id === typeAndTheme?.type?.id)
-      ?.otherSuggestedTerms ?? [];
+  const themes = useMemo(
+    () =>
+      types?.find((type) => type.id === typeAndTheme?.type?.id)
+        ?.otherSuggestedTerms ?? [],
+    [typeAndTheme?.type?.id, types],
+  );
+
+  const shouldGroupThemes = [
+    '0.3.1.0.1', // Cursus met open sessies
+    '0.3.1.0.0', // Lessenreeks
+  ].includes(typeAndTheme?.type?.id);
+
+  const themeGroups = useMemo(() => {
+    if (!shouldGroupThemes) return {};
+    return groupBy(themes, (theme) => {
+      const foundGroupPair = Object.entries(groupNameToThemeIds).find(
+        ([_, themeIds]) => themeIds.includes(theme.id),
+      );
```

Comment:
If only the value is used I'm not sure you need to transform them to entries first

Reply:
The key is also used below this code. 
I need the values (the `themeIds`) to search the correct pair. Then I return the key corresponding to the matching `themeIds`. 


While I don't use the label/key (e.g. "dance") in the find function. I do return it after the correct entry is found.

e.g.:
If the `theme.id` matches one of the ids in `DANCE_THEME_IDS`, I find that pair (["dance", DANCE_THEME_IDS]) and then return the first index of that pair to the groupBy function ("dance"). The `groupBy` function uses that returned label as a label to group the matching themes under.